engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -479,9 +499,16 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     ()
   }
 
-  private def saveCacheResults(hashes: CallCacheHashes, data: SucceededResponseData) = {
-    callCacheWriteActor ! SaveCallCacheHashes(workflowIdForLogging, hashes, data)
-    val updatedData = data.copy(hashes = Option(Success(hashes)))
+  private def saveCacheResults(hashes: CallCacheHashes, data: CacheWriteResponseData) = {
+    data match {
+      case jobSucceededData: SucceededResponseData =>
+        callCacheWriteActor ! SaveCallCacheHashes(CallCacheHashBundle(workflowIdForLogging, hashes, jobSucceededData.response))
+      case jobFailedNonRetryableData: FailedNonRetryableResponseData =>
+        callCacheWriteActor ! SaveCallCacheHashes(CallCacheHashBundle(workflowIdForLogging, hashes, jobFailedNonRetryableData.response))
```

Comment:
yes, except that `SucceededResponseData` and `FailedNonRetryableResponseData` both override their `response` field with a refined type of `BackendJobExecutionResponse` (respectively `JobSucceededResponse` and `JobFailedNonRetryableResponse`) which are the only ones a `CallCacheHashBundle` can be built from.  This avoids having to create yet another sealed trait that only `JobSucceededResponse` and `JobFailedNonRetryableResponse` extend from.